luxtronik/__init__.py

```diff
@@ -38,31 +39,46 @@ def is_socket_closed(sock: socket.socket) -> bool:
     """Check is socket closed."""
     try:
         # this will try to read bytes without blocking and also without removing them from buffer
-        data = sock.recv(
-            LUXTRONIK_SOCKET_READ_SIZE_PEEK, socket.MSG_DONTWAIT | socket.MSG_PEEK
-        )
+        data = sock.recv(LUXTRONIK_SOCKET_READ_SIZE_PEEK, socket.MSG_DONTWAIT | socket.MSG_PEEK)
         if len(data) == 0:
             return True
     except BlockingIOError:
         return False  # socket is open and reading from it would block
     except ConnectionResetError:  # pylint: disable=broad-except
         return True  # socket was closed for some other reason
     except Exception as err:  # pylint: disable=broad-except
-        LOGGER.exception(
-            "Unexpected exception when checking if socket is closed", exc_info=err
-        )
+        LOGGER.exception("Unexpected exception when checking if socket is closed", exc_info=err)
         return False
     return False
 
 
-class Luxtronik:
-    """Main luxtronik class."""
+class LuxtronikData:
+    """
+    Collection of parameters, calculations and visiblities.
+    Also provide some high level access functions to their data values.
+    """
 
-    def __init__(self, host, port=LUXTRONIK_DEFAULT_PORT, safe=True):
+    def __init__(self, parameters, calculations, visibilities):
+        self.parameters = parameters
+        self.calculations = calculations
+        self.visibilities = visibilities
+
+    @classmethod
+    def new(cls, safe=True):
+        return LuxtronikData(Parameters(safe), Calculations(), Visibilities())
+
+    @classmethod
+    def copy(cls, data):
```

Comment:
No longer needed?

Reply:
This would be necessary if the user wanted to save “old” Luxtronik Data instances.
The he could write `old_data = LuxtronikData.copy(lux.read())`